tests/pos/i3976.scala

```diff
@@ -0,0 +1 @@
+enum class Hoge[F[_]]
```

Comment:
I would add a test with an `M[_[_]]` param, and perhaps some code that exercises the generated Eq implicits too to make sure it works as expected.